tests/intl/intltest.cpp

```diff
@@ -526,6 +526,124 @@ TEST_CASE("wxLocaleIdent::FromTag", "[uilocale][localeident]")
+    CHECK(FindBestMatch({ "ja-Jpan", "ru" }, { "ja", "ja-Jpan", "ja-JP", "en", "ru" }, "en-GB") == "ja-Jpan");
+
+    // match on maximized tag
+    CHECK(FindBestMatch({ "ja-JP", "en-GB" }, { "fr", "en-GB", "ja", "es-ES", "es-MX" }, "en-US") == "ja");
+    CHECK(FindBestMatch({ "ja-Jpan-JP", "en-GB" }, { "fr", "en-GB", "ja", "es-ES", "es-MX" }, "en-US") == "ja");
+
+    // region distance German
+    CHECK(FindBestMatch({ "de" }, { "de-AT", "de-DE", "de-CH" }, "en-GB") == "de-DE");
+
+    // en-AU is closer to en-GB than to en (which is en-US)
+    CHECK(FindBestMatch({ "en-AU" }, { "en", "en-GB", "es-ES" }, "") == "en-GB");
+
+    // if no preferred locale specified, pick top language, not regional
+    CHECK(FindBestMatch({ "fr-US" }, { "en", "fr", "fr-CA", "fr-CH"}, "en-GB") == "fr");
+
+    // return most originally similar among likely-subtags equivalent locales
+    CHECK(FindBestMatch({ "af" }, { "af", "af-Latn", "af-Arab" }, "en-GB") == "af");
+    CHECK(FindBestMatch({ "af-ZA" }, { "af", "af-Latn", "af-Arab" }, "en-GB") == "af");
+    CHECK(FindBestMatch({ "af-Latn-ZA" }, { "af", "af-Latn", "af-Arab" }, "en-GB") == "af-Latn");
+    CHECK(FindBestMatch({ "af-Latn" }, { "af", "af-Latn", "af-Arab" }, "en-GB") == "af-Latn");
+
+    CHECK(FindBestMatch({ "nl" }, { "nl", "nl-NL", "nl-BE" }, "en-GB") == "nl");
+    CHECK(FindBestMatch({ "nl-Latn" }, { "nl", "nl-NL", "nl-BE" }, "en-GB") == "nl");
+    CHECK(FindBestMatch({ "nl-Latn-NL" }, { "nl", "nl-NL", "nl-BE" }, "en-GB") == "nl-NL");
+    CHECK(FindBestMatch({ "nl-NL" }, { "nl", "nl-NL", "nl-BE" }, "en-GB") == "nl-NL");
+
+    CHECK(FindBestMatch({ "nl" }, { "nl", "nl-Latn", "nl-NL", "nl-BE" }, "en-GB") == "nl");
+    CHECK(FindBestMatch({ "nl-Latn" }, { "nl", "nl-Latn", "nl-NL", "nl-BE" }, "en-GB") == "nl-Latn");
+    CHECK(FindBestMatch({ "nl-NL" }, { "nl", "nl-Latn", "nl-NL", "nl-BE" }, "en-GB") == "nl-NL");
+    CHECK(FindBestMatch({ "nl-Latn-NL" }, { "nl", "nl-Latn", "nl-NL", "nl-BE" }, "en-GB") == "nl-Latn");
+
+    // nearby languages: Danish matches no
+    CHECK(FindBestMatch({ "da" }, { "en", "no" }, "en-GB") == "no");
+    // nearby languages: Nynorsk to Bokmål
+    CHECK(FindBestMatch({ "nn" }, { "en", "nb" }, "en-GB") == "nb");
+    // nearby languages: Danish does not match nn
+    CHECK(FindBestMatch({ "da" }, { "en", "nn" }, "en-GB") == "en-GB");
+
+    // script fallbacks
+    CHECK(FindBestMatch({ "zh-Hant" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-TW");
+    CHECK(FindBestMatch({ "zh" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-CN");
+    CHECK(FindBestMatch({ "zh-Hans-CN" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-CN");
+    CHECK(FindBestMatch({ "zh-Hant-HK" }, { "zh-CN", "zh-TW" }, "en-GB") == "zh-TW");
+
+    // language-specific script fallbacks
+    CHECK(FindBestMatch({ "sr" }, { "en", "sr-Latn" }, "en-GB") == "sr-Latn");
+    CHECK(FindBestMatch({ "sr-Cyrl" }, { "en", "sr-Latn" }, "en-GB") == "sr-Latn");
```

Comment:
Also should test that `sr-Latn` is correctly handled, including (a) on Windows where IIRC you said `sr` is considered `sr-Latn` (while it is `sr-Cyrl` on Linux) and that this correctly handles the situation where the available list has Linux-style `sr` (Cyrillic) and `sr@latin` translation files.

Reply:
> Also should test that `sr-Latn` is correctly handled, including (a) on Windows where IIRC you said `sr` is considered `sr-Latn` (while it is `sr-Cyrl` on Linux) and that this correctly handles the situation where the available list has Linux-style `sr` (Cyrillic) and `sr@latin` translation files.

What I said about handling of `sr` on Windows, is related to locales. For matching translation languages we use the CLDR data, where `sr` maps to `sr-Cyrl`. Nevertheless, I add a test case.

Reply:
> What I said about handling of sr on Windows, is related to locales. For matching translation languages we use the CLDR data, where sr maps to sr-Cyrl. Nevertheless, I add a test case.

Thanks! I think the overall theme of these is that I would be more comfortable with having strong tests that go through typical scenarios for translations. For `sr`, it is relevant what the OS returns as user preference and that wx interprets it correctly. If it never returns plain `sr`, then there's no issue, of course. 

On a related note, it would be nice to have the tests go through POSIX handling too — in the available list, to make sure enumerations from `wxFilesTranslationsLoader` are parsed correctly.

Reply:
> I think the overall theme of these is that I would be more comfortable with having strong tests that go through typical scenarios for translations. For `sr`, it is relevant what the OS returns as user preference and that wx interprets it correctly. If it never returns plain `sr`, then there's no issue, of course.

Unfortunately, I was not able to test this under Windows 11. Although I installed "Serbian (Serbia), Latin" and "Serbian (Serbia), Cyrillic" as additional preferred languages in the language and region settings, the Windows system function `GetUserPreferredUILanguages` (which is called by wxUILocale internally) returns only 2 entries, namely `de-DE` and `en-US` - that is, my primary UI language German plus US-English.

However, I'm pretty sure that Windows will return `sr-Latn-SR` or `sr-Cyrl-SR`, because you have to select a language, script and region for Serbian, when you install the additional languages.

Nevertheless, in the list of known locales Windows returns also the locale identifier `sr` - and that is associated with latin script.

> On a related note, it would be nice to have the tests go through POSIX handling too — in the available list, to make sure enumerations from `wxFilesTranslationsLoader` are parsed correctly.

Method `wxLocaleIdent::FromTag()` accepts all supported locale identifier formats. And I'm pretty confident that this method works correctly. Nevertheless, you are welcome to name some typical examples for the list of available translations, and I will add tests for them.